fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneRecordCursor.java

```diff
@@ -128,141 +134,29 @@ class LuceneRecordCursor implements BaseCursor<IndexEntry> {
                 throw new RecordCoreException("Invalid continuation for Lucene index", "ContinuationValues", continuation, e);
             }
         }
-        this.currentPosition = 0;
-        if (scanProperties.getExecuteProperties().getSkip() > 0) {
-            this.currentPosition += scanProperties.getExecuteProperties().getSkip();
-        }
         this.fields = state.index.getRootExpression().normalizeKeyForPositions();
         this.groupingKey = groupingKey;
     }
 
     @Nonnull
     @Override
     public CompletableFuture<RecordCursorResult<IndexEntry>> onNext() {
-        if (nextResult != null && !nextResult.hasNext()) {
-            // Like the KeyValueCursor, it is necessary to memoize and return the first result where
-            // hasNext is false to avoid the NoNextReason changing.
-            return CompletableFuture.completedFuture(nextResult);
```

Comment:
I think we want to keep this code that memoizes the `nextResult` once we hit a reason that stops the cursor from returning data.

Reply:
Done